lib/debug.py

```diff
@@ -0,0 +1,197 @@
+        #    * follow the "Note:" to enable the debug server menu item
+        # 3. Create a file named "DEBUG" next to inkstitch.py in your git clone.
+        # 4. Run any extension and PyDev will start debugging.
+
+        try:
+            import pydevd
+        except ImportError:
+            self.log("importing pydevd failed (debugger disabled)")
+
+        # pydevd likes to shout about errors to stderr whether I want it to or not
+        with open(os.devnull, 'w') as devnull:
+            stderr = sys.stderr
+            sys.stderr = devnull
+
+            try:
+                pydevd.settrace()
+            except socket.error, error:
+                self.log("Debugging: connection to pydevd failed: %s", error)
+                self.log("Be sure to run 'Start debugging server' in PyDev to enable debugging.")
+            else:
+                self.log("Enabled PyDev debugger.")
+
+            sys.stderr = stderr
+
+    def init_svg(self):
+        self.svg = etree.Element("svg", nsmap=inkex.NSS)
+        atexit.register(self.save_svg)
+
+    def save_svg(self):
+        tree = etree.ElementTree(self.svg)
+        with open(os.path.join(os.path.dirname(os.path.dirname(__file__)), "debug.svg"), "w") as debug_svg:
+            tree.write(debug_svg)
+
+    @check_enabled
+    def add_layer(self, name="Debug"):
+        layer = etree.Element("g", {
+            INKSCAPE_GROUPMODE: "layer",
+            INKSCAPE_LABEL: name,
+            "style": "display: none"
+        })
+        self.svg.append(layer)
+        self.current_layer = layer
+
+    @check_enabled
+    def open_group(self, name="Group"):
+        group = etree.Element("g", {
+            INKSCAPE_LABEL: name
+        })
+
+        self.log_svg_element(group)
+        self.group_stack.append(group)
+
+    @check_enabled
+    def close_group(self):
+        if self.group_stack:
+            self.group_stack.pop()
+
+    @check_enabled
+    def log(self, message, *args):
+        if self.last_log_time:
+            message = "(+%s) %s" % (datetime.now() - self.last_log_time, message)
+
+        self.raw_log(message, *args)
+
+    def raw_log(self, message, *args):
+        now = datetime.now()
+        timestamp = now.isoformat()
+        self.last_log_time = now
+
+        print >> self.log_file, timestamp, message % args
+        self.log_file.flush()
+
+    def time(self, func):
+        def decorated(*args, **kwargs):
+            if self.enabled:
+                self.raw_log("entering %s()", func.func_name)
+                start = time.time()
+
+            result = func(*args, **kwargs)
+
+            if self.enabled:
+                end = time.time()
+                self.raw_log("leaving %s(), duration = %s", func.func_name, round(end - start, 6))
+
+            return result
+
+        return decorated
+
+    @check_enabled
+    def log_svg_element(self, element):
+        if self.current_layer is None:
```

Comment:
I think it's more pythonic to check for none without the ``is None``` unless self.current_layer will sometimes be 0 or an empty string

Reply:
Oh, this one's a little tricky.  `self.current_layer` is an lxml element tag.  Element instances act like lists and you can index them, slice, append, etc.  

If you try to treat one as a boolean directly, it complains at you with a deprecation warning because it's ambiguous whether you mean None or an XML tag with no child tags.  So we have to compare to None in this case.